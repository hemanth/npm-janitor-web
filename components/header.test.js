import React from 'react'
import Header from './header'
import renderer from 'react-test-renderer'

jest.mock('next/router')

it('renders correctly', () => {
  let props = {full: false}
  const component = renderer.create(<Header {...props}/>)
  let tree = component.toJSON()

  let input = tree.children[0].children[1].children[0]
  input.props.onChange({target: {value: 'siddharthkp'}})
  input.props.onKeyUp({which: 65})
  input.props.onKeyUp({which: 13})

  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
