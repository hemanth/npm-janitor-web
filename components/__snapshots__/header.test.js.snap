// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="XRAxu"
>
  <div>
    <img
      className="hQbxwO"
      src="static/logo.png"
    />
    <div>
      <input
        autoFocus={true}
        className="ftiIRS"
        defaultValue={undefined}
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="enter your npm username"
        type="text"
      />
      <div
        className="bvUXDN"
      />
    </div>
  </div>
</div>
`;
